tests/e2e/specFiles/Transactions/receive.spec.ts

```diff
@@ -0,0 +1,135 @@
+  DEFAULT_VALUE: 'Amount',
+  AMOUNT_TO_RECEIVE: '10',
+};
+
+describe('Transactions', () => {
+  beforeEach(async () => {
+    isBeta() && (await app.onboarding.betaVersionScreen.close());
+    await app.developerRoom.tapOnSkipOnboardingButton();
+    await app.navigationBar.changeTab('wallets');
+  });
+
+  describe('Receive', () => {
+    describe('3-Key Vault', () => {
+      describe('@android @iOS @smoke', () => {
+        it('should be possible to see QRCode, wallet address and receive amount for 3-Key Vault wallet', async () => {
+          await app.wallets.createWallet({
+            type: '3-Key Vault',
+            name: '3-Key wallet',
+            fastPublicKey: WALLETS_WITH_COINS['3-Keys Vault'].FAST_KEY,
+            cancelPublicKey: WALLETS_WITH_COINS['3-Keys Vault'].CANCEL_KEY,
+          });
+          await app.wallets.dashboardScreen.tapOnReceiveButton();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.qrCodeIcon).toBeVisible();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.walletAdressText).toBeVisible();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.receiveAmountText).toHaveText(
+            DATA_FOR_TRANSACTIONS.DEFAULT_VALUE,
+          );
+          await app.transactionsReceive.receiveCoinsMainScreen.typeAmountToReceive(
+            DATA_FOR_TRANSACTIONS.AMOUNT_TO_RECEIVE,
+          );
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.receiveAmountText).toHaveText(
+            DATA_FOR_TRANSACTIONS.AMOUNT_TO_RECEIVE,
+          );
+        });
+      });
+
+      describe('@android @iOS @regression', () => {
+        it('should be possible to see QRCode, wallet address and receive amount for 2-Key Vault wallet', async () => {
+          await app.wallets.createWallet({
+            type: '2-Key Vault',
+            name: '2-Key wallet',
+            cancelPublicKey: WALLETS_WITH_COINS['2-Keys Vault'].CANCEL_KEY,
+          });
+          await app.wallets.dashboardScreen.tapOnReceiveButton();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.qrCodeIcon).toBeVisible();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.walletAdressText).toBeVisible();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.receiveAmountText).toHaveText(
+            DATA_FOR_TRANSACTIONS.DEFAULT_VALUE,
+          );
+          await app.transactionsReceive.receiveCoinsMainScreen.typeAmountToReceive(
+            DATA_FOR_TRANSACTIONS.AMOUNT_TO_RECEIVE,
+          );
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.receiveAmountText).toHaveText(
+            DATA_FOR_TRANSACTIONS.AMOUNT_TO_RECEIVE,
+          );
+        });
+      });
+    });
+
+    describe('Standard HD P2SH', () => {
+      describe('@android @iOS @regression', () => {
+        it('should be possible to see QRCode, wallet address and receive amount for Standard HD P2SH wallet', async () => {
```

Comment:
I really doubt if it brings any value to repeat the same test for every wallet type

Reply:
I had spoken with @KedziaPawel about this and the reason behind that is that we are using func to get wallet address for transaction and it is different for each type of wallet.